### PR TITLE
add functionality to post comment, WIP for receiving error feedback

### PR DESCRIPTION
**Pull request by vh232**

### CODE REVIEW

File: src/components/AddAComment.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const { article_id } = useParams();
+  const { comments, setComments } = props;
+  const [commentBody, setCommentBody] = useState();
+  const [newComment, setNewComment] = useState({
+    username: "grumpy19",
+    body: "",
+  });
+
+  const focusOnCommentBox = () => {
+    document.getElementById("add-comment").focus();
```
Comment (saimakhanom) [inline src/components/AddAComment.jsx:17]: check out the onFocus attribute for a simpler way to do this

File: src/components/AddAComment.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  };
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+    setNewComment({ ...newComment }, (newComment.body = commentBody));
```
Comment (saimakhanom) [inline src/components/AddAComment.jsx:22]: good optimistic rendering

File: src/components/AddAComment.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    postComment(article_id, newComment).then((newCommentFromAPI) => {
+      if (newCommentFromAPI.code) {
+
```
Comment (saimakhanom) [inline src/components/AddAComment.jsx]: not sure whats supposed to happen here?

Comment (vh232) [reply src/components/AddAComment.jsx]: Whoops honestly no idea, think that was from me trying to do error handling then I decided to do it during task 12 instead - deleted!

**Pull request merged**